### PR TITLE
[Debt] Replace `useLocale` hook

### LINKED ISSUE #10133: ♻️ Remove `useLocale` in pages and components
## ♻️ Refactor

Remove `useLocale` in pages and components.

## 🕵️ Details

> we probably shouldn't be using this hook in normal pages/components 😬 This hook is meant to get and set the locale for the `IntlProvider`  and `Router`.

REF: https://github.com/GCTC-NTGC/gc-digital-talent/issues/8005#issuecomment-2047380159

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] no instances of `useLocale` in pages and components

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #10133 

## 👋 Introduction

Replaces the `useLocale` hook with the `getLocale` helper in pages and components.

## 🕵️ Details

The `useLocale` is for use above the `LanguageProvider` or, setting the locale. This replaces it in any place it was being used outside of those use cases.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `pnpm run dev`
2. Confirm language switching still works for the places where locale was updated
3. Confirm `useLocale` only appears in `Header`, `Router` and context providers

### CODE REVIEW

**Comment by vd1992** [comment]
I think this bit is missing

image

File: packages/forms/src/components/ErrorSummary.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const numberRegex = /(\d+)/g;
 /**
  * Get Field Label
  *
- * User the field name to extract the associated label
+ * , getLocaleUser the field name to extract the associated label
```
Comment (mnigh) [inline packages/forms/src/components/ErrorSummary.tsx:32]: A case of find and replace gone wrong?

**Review by yonikid15 — APPROVED** [review]
Looks good 👍

**Comment by mnigh** [comment]
[Linter](https://github.com/GCTC-NTGC/gc-digital-talent/actions/runs/8833010364/job/24251427456?pr=10161) caught one more instance.

## Screenshot

Screen Shot 2024-04-25 at 10 13 39

**Pull request merged**